app/src/androidTest/java/ch/epfl/sdp/peakar/points/ComputePOIPointsTest.java

```diff
@@ -87,24 +88,23 @@ public void computeVerticalBearing(){
     }
 
     /**
-     * Test if ComputePOIPoints can get POIPoints and LabeledPOIPoints
-     * @throws InterruptedException if any thread has interrupted the current thread
+     * Test if ComputePOIPoints has computed the POIs
      */
     @Test
-    public void getLabeledPOIPoints() throws InterruptedException {
-        Context mContext = ApplicationProvider.getApplicationContext();
-
-        new ComputePOIPoints(mContext);
-
-        int count=0;
-        //Wait for the map to be downloaded
-        while(count<20 && ComputePOIPoints.labeledPOIPoints==null){
-            Thread.sleep(1000);
-            count++;
-        }
-
-        //Check that POIPoints and labeledPOIPoints are not null
-        assertNotNull(ComputePOIPoints.POIPoints);
-        assertNotNull(ComputePOIPoints.labeledPOIPoints);
+    public void getPOIPointsTest() {
+        assertNotNull(ComputePOIPoints.getPOIs());
+        assertNotNull(ComputePOIPoints.getPOIsInSight());
+        assertNotNull(ComputePOIPoints.getPOIsOutOfSight());
+        assertNotNull(ComputePOIPoints.getFilteredPOIs());
+        assertNotNull(ComputePOIPoints.getFilteredPOIsInSight());
+        assertNotNull(ComputePOIPoints.getFilteredPOIsOutOfSight());
+
+        assertTrue(ComputePOIPoints.getPOIs().size() > 0);
+        assertEquals(ComputePOIPoints.getPOIs().size(), ComputePOIPoints.getPOIsInSight().size()
+                + ComputePOIPoints.getPOIsOutOfSight().size());
+
+        assertTrue(ComputePOIPoints.getFilteredPOIs().size() > 0);
```

Comment:
Here we should maybe check if the POIs returned by the getPOIs methods are actually the one we expect to see. A comparison with a constant list of POI would be nice to have. Same remark for all `getXXXPOIsXXX()` methods used in this test.